protocol.html

```diff
@@ -476,6 +480,14 @@ <h3 property="schema:name">Storage</h3>
 
                   <p>[<a href="https://github.com/solid/data-interoperability-panel/issues/10#issuecomment-598694029" rel="cito:citesAsSourceDocument">Source</a>] [<a href="https://github.com/solid/specification/issues/153#issuecomment-624630022" rel="cito:citesAsSourceDocument">Source</a>]</p>
 
+                  <p>Servers MUST keep track of at least one <a data-link-type="dfn" href="#owner" id="ref-for-owner">owner</a> of a storage in an implementation defined way.</p>
+
+                  <p>When a server wants to advertise the owner of a storage, the server MUST include the <code>Link</code> header with <code>rel="http://www.w3.org/ns/solid/terms#owner"</code> targeting the URI of the owner in the response of HTTP <code>HEAD</code> or <code>GET</code> requests targeting the root container.</p>
```

Comment:
> When a server wants to advertise the owner ...

Does this allow for the possibility that an owner may wish to remain anonymous? Another way to phrase this question: is it up to the server to decide whether to advertise the owner? Or is that an implementation detail?

Reply:
[Good question. Can of worms on data minimisation...]

There is no constraint on reference to owner entailing a real, anonymous or pseudonymous identity. There is also no requirement (so far) that the URI of the owner needs to have a representation. While information could be inferred from the URI, and as you know, ultimately it is opaque. So, the URI doesn't tell us much about the social entity behind it. I also don't think the owner link relation should be coupled with access rights.

No need make the distinction in the spec about the owner's identity, anonymity, pseudonymity, as there are good cases for all; implementation detail. One side-effect of that is that identities could also be linkable by the system eg. the anonymous owner with the real identity.

The main question may just boil down to whether the owner must always be discoverable or not. If the spec requires owners to be discoverable, an implementation could for example choose to use an anonymous identifier, which may effectively be same as not advertising it to begin with, especially if the owner resource has no representation or no access provided to the world.

I'd be comfortable with the idea/requirement that  server always advertises the owner link relation. WDYT?

Reply:
Great question indeed! It seems that in the interest of data minimisation, it shouldn't be a requirement to make it discoverable, and I don't see the usefulness of an anonymous identifier. A pseudonym could be useful though.

Generally, we should assume that a WebID is personally identifiable information (because it really is, even though good pseudonymisation should help with that). 

I imagine the exposure of an owner should be configurable, and so, yes, an implementation detail. I think we should note that in the privacy considerations in the spec though, so people aren't just making assumptions about it.

Reply:
The usefulness of any identifier is that, if it resolves to a representation, the owner can be contacted eg. via its inbox, without reveal anything else.

Reply:
right, but any attribute might also reveal identity... Can we even talk about an anonymous identity, aren't we then really talking about a pseudonym? 

I think a sophisticated treatment of pseudonyms is a very interesting topic, but not something we should do before this is merged.

Reply:
Basically, you have to ensure that pseudonymity or anonymity is maintained with all subsequent operations, i.e. you dereference the owner's URI, you get back some attributes amongst those, an inbox. 

With anonymity, you have to ensure that none of these things can tell you anything at all. With pseudonymity, you just have to ensure that these things cannot tell someone anything more than they already know about the identity. It is probably easier. I think... :-)

Anyway, there seems to be two open issues here, the rest can go in. How about this: We merge this PR as it currently is, and then we open two issues: 

- [ ] What is the domain of the `solid:owner` relation?
- [ ] whether the owner must be discoverable or not.

Reply:
Thanks, @csarven and @kjetilk. In the context of my question, I think it's possible to set aside the question of identifier resolution (n.b. this may be an interesting use case for DIDs, so for the domain/range conversation, that would be something to consider).

> I'm leaning on requiring [the owner identifier relation] without expecting anything further about the identity that it refers to.

:+1: I would support that.

Reply:
I'm :-1: because there is an ocean of difference between anonymity and pseudonymity when you come to legal matters, for example GDPR applies to pseudonyms but not when the user is anonymous. Lets merge this and open new issues.

Reply:
The Protocol is not prescribing the nature of the owner URI, and leaves it as an implementation detail that could be configurable to be compliant with what's legally applicable at pod provisioning time (by its owner for example).

No, the discovery of the owner link relation - server-controlled HTTP header - is not coupled with access privileges.

Reply:
> The Protocol is not prescribing the nature of the owner URI, and leaves it as an implementation detail that could be configurable to be compliant with what's legally applicable at pod provisioning time (by its owner for example).

Still, lets open a separate issue on it. The use cases are really unclear to me, and the requirements around what is an anonymous identifier (is there such a thing?) and what you'd get out of dereferencing it is also unclear.

Reply:
>The use cases are really unclear to me

System enforcing owner. System enabling clients to discover owner.

>the requirements around what is an anonymous identifier (is there such a thing?) and what you'd get out of dereferencing it is also unclear.

That's because this PR did not set out to prescribe what the owner identifier should look like (or imply anything beyond it being opaque) or whether the system requires to generate a representation and what the description needs to look like. Orthogonal.

Sure, we can continue/extend the discussion in the issues as you've created but I didn't see these as blockers for this PR (FWIW).

Reply:
Yup, and exactly because it is orthogonal, it should have a separate issue. :-)